Src/ILGPU/Backends/PTX/PTXCodeGenerator.cs

```diff
@@ -234,7 +234,18 @@ public PhiBindingAllocator(PTXCodeGenerator parent)
                 default, BasicValueType.Int1,
                 BasicValueType.Int16, BasicValueType.Int16,
                 BasicValueType.Int32, BasicValueType.Int64,
-                BasicValueType.Float32, BasicValueType.Float64);
+                BasicValueType.Int16, BasicValueType.Float32, BasicValueType.Float64);
+
+        /// <summary>
+        /// Maps basic types to constant-loading target basic types.
+        /// </summary>
+        private static readonly ImmutableArray<BasicValueType>
+            RegisterIOTypeRemapping =
+            ImmutableArray.Create(
+                default, BasicValueType.Int8,
+                BasicValueType.Int8, BasicValueType.Int16,
+                BasicValueType.Int32, BasicValueType.Int64,
+                BasicValueType.Int16, BasicValueType.Float32, BasicValueType.Float64);
```

Comment:
```suggestion
                BasicValueType.Float16, BasicValueType.Float32, BasicValueType.Float64);
```

Reply:
This change should be correct since we have to map `fp16` values to `int16` registers and we cannot represent `fp16` constants in PTX. Consequently, ILGPU loads the raw value representation of a `fp16` into a 16bit int register.

Reply:
ah, I see 👍